### PR TITLE
Fix a potential crash when initializing the map for meshes with holes

### PR DESCRIPTION
**Pull request by martin-marinov77**
* Fix a potential crash when initializing the map for meshes with holes
* Fix a minor conversion warning

Without this change we can get crashes on models with holes.

### CODE REVIEW

**Comment by jiangzhongshi** [comment]
@jdumas After changing the base branch, some more commits are appended. Is there an easy way from our side to dismiss them?

**Comment by martin-marinov77** [comment]
I can rebase this on top of dev and resubmit if that makes your life easier?
I did the change on master only because it's a crash fix and people consuming libigl might want to get this immediately rather than wait for dev to get merged into master...

**Comment by jdumas** [comment]
If people want the fix they should use the dev branch. You can rebase onto dev and force-push into this PR, no need to create a new PR.

**Comment by martin-marinov77** [comment]
OK, done.

**Comment by jiangzhongshi** [comment]
Hi Martin,

Thanks for the PR, but I don't quite understand why it would potentially crash, at least on my desktop. Maybe I missed something, can you point it out?
I admit this is not the best way to write it. `uv_init.conservativeResize(V.rows(), 2)` could be a more succinct way.

**Comment by martin-marinov77** [comment]
This did crash in Eigen on some of the _large_ models we've tried, and corrupted the init uv data in other cases. I did not dig into it too deep, but I did see Eigen doing some weird stuff internally in `uv_init = uv_init.topRows(V.rows())`. I cannot share the test models, but maybe i can get a call-stack to attach to this to convince you. The resize approach might work as well, I did not think of it.

Bottom line is that after I've replaced `uv_init = uv_init.topRows(V.rows())` with the statements in the commit, everything started to work...

**Comment by martin-marinov77** [comment]
Here you go, this is what I get with a large mesh with many holes and a very large outer boundary. 

image

I will not be surprised if Eigen somehow decides to free the existing memory in `uv_init` in the assignment operator and allocate new storage before the copy, possibly to save some memory. Then the wrapper of `topRows()` would be reading discarded memory which may (or may not) work.

**Comment by martin-marinov77** [comment]
`uv_init.conservativeResize(V.rows(), 2)` is better indeed, it does the same (free, alloc and copy) but it's more concise. So I've pushed another commit with the change. Thanks for tip.
